vms/avm/wallet_service_backend.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package avm
+
+import (
+	"context"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/linked"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/avm/txs"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/builder"
+)
+
+var _ txBuilderBackend = (*walletServiceBackend)(nil)
+
+func NewWalletServiceBackend(vm *VM) *walletServiceBackend {
```

Comment:
Can we accept an `UTXOReader` instead of the concrete `*VM` type here?

Reply:
the vm here is used by wallet service as well. I'd keep vm to centralize access